hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -265,6 +268,27 @@ public ContainerStateMachine(HddsDatanodeService hddsDatanodeService, RaftGroupI
 
   }
 
+  private void validatePeers(RaftServer server, RaftGroupId id) throws IOException {
+    if (this.peersValidated.get()) {
+      return;
+    }
+    synchronized (peersValidated) {
```

Comment:
Do you think synchronized is necessary here? Or would it make sense to use volatile instead of AtomicBoolean for peersValidated?

Reply:
I removed the synchronized block